examples/cw-contract/src/contract.rs

```diff
@@ -106,3 +106,127 @@ fn query_fetch_price(deps: Deps, env: Env) -> StdResult<FetchPriceResponse> {
         ema_price,
     })
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use cosmwasm_std::testing::{
+        mock_dependencies,
+        mock_env,
+        MockApi,
+        MockQuerier,
+        MockStorage,
+    };
+    use cosmwasm_std::{
+        from_binary,
+        Addr,
+        Coin,
+        OwnedDeps,
+        QuerierResult,
+        SystemError,
+        SystemResult,
+        Timestamp,
+        WasmQuery,
+    };
+    use pyth_sdk_cw::test_utils::MockPyth;
+    use pyth_sdk_cw::{
+        Price,
+        PriceFeed,
+        PriceIdentifier,
+        UnixTimestamp,
+    };
+    use std::convert::TryFrom;
+    use std::time::Duration;
+
+    // Dummy contract address for testing.
+    // For real deployments, see list of contract addresses here https://docs.pyth.network/pythnet-price-feeds/cosmwasm
+    const PYTH_CONTRACT_ADDR: &str = "pyth_contract_addr";
+    // For real deployments, see list of price feed ids here https://pyth.network/developers/price-feed-ids
+    const PRICE_ID: &str = "63f341689d98a12ef60a5cff1d7f85c70a9e17bf1575f0e7c0b2512d48b1c8b3";
+
+    fn default_state() -> State {
+        State {
+            pyth_contract_addr: Addr::unchecked(PYTH_CONTRACT_ADDR),
+            price_feed_id:      PriceIdentifier::from_hex(PRICE_ID).unwrap(),
+        }
+    }
+
+    fn setup_test(
+        state: &State,
+        mock_pyth: &MockPyth,
+        block_timestamp: UnixTimestamp,
+    ) -> (OwnedDeps<MockStorage, MockApi, MockQuerier>, Env) {
+        let mut dependencies = mock_dependencies();
+
+        let mock_pyth_copy = (*mock_pyth).clone();
+        dependencies
+            .querier
+            .update_wasm(move |x| handle_wasm_query(&mock_pyth_copy, x));
+
+        STATE.save(dependencies.as_mut().storage, state).unwrap();
+
+        let mut env = mock_env();
+        env.block.time = Timestamp::from_seconds(u64::try_from(block_timestamp).unwrap());
+
+        (dependencies, env)
+    }
+
+    // Create a handler like this in your test to handle pyth queries. If needed, other contracts
+    // can be configured in this handler via additional cases.
+    fn handle_wasm_query(pyth: &MockPyth, wasm_query: &WasmQuery) -> QuerierResult {
```

Comment:
couldn't we move this method to the mock contract itself? it seems to be related to the mock contract (maybe it gets the mock dependency on ::new and handle it there?)

Reply:
this is actually the handler for all cross-contract calls. if the user has multiple contracts to dispatch to, they will need to add them as cases here.

Reply:
right. thanks for explaining.